uitools/toolkitcpp/cpp/Esri/ArcGISRuntime/Toolkit/MediaPopupElementViewController.cpp

```diff
@@ -48,9 +55,40 @@ MediaPopupElementViewController::MediaPopupElementViewController(
   : PopupElementViewItem{mediaPopupElement, parent},
     m_popupMediaItems{new GenericListModel(&PopupMediaItem::staticMetaObject, this)}
 {
-  for (auto* media: *static_cast<MediaPopupElement*>(popupElement())->media())
+  auto* media = mediaPopupElement->media();
+  for (int i = 0; i < media->size(); i++)
   {
-    m_popupMediaItems->append(new PopupMediaItem(media, this));
+    const QJsonDocument mediaPopupElementJson = QJsonDocument::fromJson(mediaPopupElement->toJson().toUtf8());
+    auto* popupMedia = media->at(i);
+
+    // Get color information for each popup media. If no color is selected when defining a popup, the default
+    // color will be used from the appropriate QtGraphs Series component.
+    // s.a. Web Map Spec: https://developers.arcgis.com/web-map-specification/objects/popupElement_media
+    const auto colors = mediaPopupElementJson.object()["mediaInfos"][i]["value"]["colors"].toArray();
+
+    switch (popupMedia->popupMediaType())
+    {
+      case Esri::ArcGISRuntime::PopupMediaType::Image:
+        m_popupMediaItems->append(new ImagePopupMediaItem(popupMedia, media));
+        break;
+      case Esri::ArcGISRuntime::PopupMediaType::BarChart:
+        m_popupMediaItems->append(new BarChartPopupMediaItem(popupMedia, colors, media));
+        break;
+      case Esri::ArcGISRuntime::PopupMediaType::ColumnChart:
+        m_popupMediaItems->append(new BarChartPopupMediaItem(popupMedia, colors, media));
+        break;
+      case Esri::ArcGISRuntime::PopupMediaType::PieChart:
+        m_popupMediaItems->append(new PieChartPopupMediaItem(popupMedia, colors, media));
+        break;
+      case Esri::ArcGISRuntime::PopupMediaType::LineChart:
+        m_popupMediaItems->append(new LineChartPopupMediaItem(popupMedia, colors, media));
+        break;
+      case Esri::ArcGISRuntime::PopupMediaType::Unknown:
+        break;
+      default:
+        Q_UNIMPLEMENTED();
+        break;
```

Comment:
Are there more types? Ideally we don't have default switch cases.

Reply:
No there are not. I'll remove this.